webapp/src/components/react_select_setting.jsx

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ReactSelect from 'react-select';
+import AsyncSelect from 'react-select/async';
+
+import Setting from 'components/setting';
+import {getStyleForReactSelect} from 'utils/styles';
+
+const MAX_NUM_OPTIONS = 100;
+
+export default class ReactSelectSetting extends React.PureComponent {
+    static propTypes = {
+        name: PropTypes.string,
+        onChange: PropTypes.func,
+        theme: PropTypes.object.isRequired,
+        isClearable: PropTypes.bool,
+        options: PropTypes.array.isRequired,
+        isLoading: PropTypes.bool,
+        value: PropTypes.oneOfType([
+            PropTypes.object,
+            PropTypes.array,
+            PropTypes.string,
+        ]),
+        addValidate: PropTypes.func,
+        removeValidate: PropTypes.func,
+        required: PropTypes.bool,
+        allowUserDefinedValue: PropTypes.bool,
+        limitOptions: PropTypes.bool,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {invalid: false};
+    }
+
+    componentDidMount() {
+        if (this.props.addValidate && this.props.name) {
+            this.props.addValidate(this.props.name, this.isValid);
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.props.removeValidate && this.props.name) {
+            this.props.removeValidate(this.props.name);
+        }
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (prevState.invalid && (this.props.value && this.props.value.value) !== (prevProps.value && prevProps.value.value)) {
+            this.setState({invalid: false}); //eslint-disable-line react/no-did-update-set-state
+        }
```

Comment:
see if we can remove `eslint-disable..` by creating a separate function for this and calling that here